registry/protocol/protocol.go

```diff
@@ -44,7 +44,7 @@ import (
 )
 
 var (
-	regProtocol *registryProtocol
+	regProtocol = newRegistryProtocol()
```

Comment:
pls do not init a package global var in this way. u should init it in a init func.

Reply:
why?

Reply:
My personal programming practice.